src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -183,15 +183,19 @@ function VisEditor($scope, $route, timefilter, AppState, $window, kbnUrl, courie
 
     $scope.timefilter = timefilter;
     $scope.opts = _.pick($scope, 'doSave', 'savedVis', 'shareData', 'timefilter', 'isAddToDashMode');
+    vis.api = {
```

Comment:
this is here so that <visualize> component doesn't need to know about this top level components directly, however it depends on them ... i am not sure if i should just move this inside <visualize> component ?